api/strategies/discordStrategy.js

```diff
@@ -0,0 +1,38 @@
+const passport = require('passport');
+const { Strategy: DiscordStrategy } = require('passport-discord');
+const User = require('../models/User');
+const config = require('../../config/loader');
+const domains = config.domains;
+
+const discordLogin = new DiscordStrategy(
+  {
+    clientID: process.env.DISCORD_CLIENT_ID,
+    clientSecret: process.env.DISCORD_CLIENT_SECRET,
+    callbackURL: `${domains.server}${process.env.DISCORD_CALLBACK_URL}`,
+    scope: ['identify', 'email'] // Request scopes
+  },
+  async (accessToken, refreshToken, profile, cb) => {
+    try {
+      const email = profile.email;
+      const oldUser = await User.findOne({ discordId: profile.id });
+      if (oldUser) {
+        return cb(null, oldUser);
+      }
+      const newUser = await User.create({
+        provider: 'discord',
+        discordId: profile.id,
+        username: profile.username,   // username is the username (the old tag discord#1111) of the user on discord
+        email,
+        name: profile.global_name,  // global_name is the name of the user on discord
+        avatar: `https://cdn.discordapp.com/avatars/${profile.id}/${profile.avatar}.png`
```

Comment:
you should handle the edge case where a user has no avatar set, and this URL should be assigned outside of this object

Reply:
I haven't tested this but according to GPT, this may help with default avatars

```js
      let avatarURL;
      if (profile.avatar) {
        const format = profile.avatar.startsWith('a_') ? 'gif' : 'png';
        avatarURL = `https://cdn.discordapp.com/avatars/${profile.id}/${profile.avatar}.${format}`;
      } else {
        const defaultAvatarNum = Number(profile.discriminator) % 5;
        avatarURL = `https://cdn.discordapp.com/embed/avatars/${defaultAvatarNum}.png`;
      }

      console.debug('Discord profile', profile);
      console.debug('Discord avatarURL', avatarURL);
```

**From GPT:**
When a user doesn't have a custom avatar, Discord provides a default one. There are five default avatars, and the one that's used for a particular user is determined by the last digit of their discriminator (the four-digit number that comes after their username), modulo 5.

Here's the relevant part of the Discord Developer Documentation that talks about this:

> "If the user has not set an avatar, the `avatar` field will be `null` and you can generate a default URL by concatenating the user's discriminator modulo 5 to the URL `https://cdn.discordapp.com/embed/avatars/.png`."

You can find this in the [User Object section](https://discord.com/developers/docs/resources/user#user-object) of the Discord Developer Documentation.

So, in the code snippet:

```javascript
const defaultAvatarNum = Number(profile.discriminator) % 5;
avatarURL = `https://cdn.discordapp.com/embed/avatars/${defaultAvatarNum}.png`;
```

`defaultAvatarNum` is the user's discriminator modulo 5, which determines which of the five default avatars to use. `avatarURL` is then the URL of this default avatar.